packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,250 @@
+	}
+
+	const minutes = parseInt(minutesString, 10);
+
+	if (minutes >= 60) {
+		const errMsg = 'TIME - INCORRECT MINUTES NUMBER';
+		return [undefined, errMsg];
+	}
+
+	const timeValidated = hoursAndMinutesToTime(hours, minutes);
+
+	return [timeValidated];
+}
+
+
+class InputDateTimePicker extends React.Component {
+	static propTypes = {
+		selectedDateTime: PropTypes.instanceOf(Date),
+		onChange: PropTypes.func,
+		onError: PropTypes.func,
+		inputProps: PropTypes.object,
+	};
+
+	constructor(props) {
+		super(props);
+
+		// eslint-disable-next-line
+		console.warn(`UNSTABLE WARNING: The 'InputDateTimePicker' and all the sub components aren't ready to be used in Apps. Code can (will) change outside the release process until it's ready.`);
+
+		this.state = (() => {
+			const selectedDateTime = this.props.selectedDateTime;
+			if (selectedDateTime !== undefined) {
+				const date = startOfDay(selectedDateTime);
+				const hours = getHours(selectedDateTime);
+				const minutes = getMinutes(selectedDateTime);
+				const time = hoursAndMinutesToTime(hours, minutes);
+				const fullDate = startOfMinute(selectedDateTime);
+
+				return {
+					date,
+					time,
+					lastFullDate: fullDate,
+					textInput: getTextDate(date, time),
+				};
+			}
+
+			return {
+				textInput: '',
+			};
+		})();
+
+		this.onChangeInput = this.onChangeInput.bind(this);
+		this.onSubmitPicker = this.onSubmitPicker.bind(this);
+	}
+
+	onSubmitPicker({ date, time }) {
+		this.updateDateTime(date, time);
+	}
+
+	onChangeInput(event) {
+		const fullString = event.target.value;
+
+		const splitMatches = fullString.match(splitDateAndTimePartsRegex);
+
+		const canParseFullString = splitMatches !== null;
+
+		const dateStrToParse = canParseFullString
+			? splitMatches[1]
+			: fullString;
+
+		const [date, errMsgDate] = extractDate(dateStrToParse);
+
+		const timeStrToParse = canParseFullString
+			? splitMatches[2]
+			: fullString;
+
+		const [time, errMsgTime] = extractTime(timeStrToParse);
+
+		const errMsg = canParseFullString
+			? errMsgDate || errMsgTime
+			: 'DATETIME - INCORRECT FORMAT';
+
+		this.updateDateTime(date, time, fullString, errMsg);
+	}
+
+	updateDateTime(date, time, textInput = getTextDate(date, time), errorMsg) {
+		const fullDate = (() => {
+			if (date === undefined || time === undefined) {
+				return undefined;
+			}
+
+			return setMinutes(date, time);
+		})();
+
+		const fullDateUpdated = fullDate !== this.state.lastFullDate
+								&& !isSameMinute(fullDate, this.state.lastFullDate);
+
+		if (fullDateUpdated && this.props.onChange) {
+			this.props.onChange(fullDate);
+		}
+
+		const errorUpdated = errorMsg !== this.state.lastErrMsg;
+		if (errorUpdated && this.props.onError) {
+			this.props.onError(errorMsg);
+		}
+
+		this.setState({
+			date,
+			time,
+			textInput,
+			lastFullDate: fullDate,
+			lastErrMsg: errorMsg,
+		});
+	}
+
+	render() {
+		const {
+			date,
+			time,
+			textInput,
+		} = this.state;
```

Comment:
No need to spread that value if it's used directly.
The `this.state.date` form is fine.

Reply:
✔